### ISSUE TITLE
[Help] - Bot setup guidelines

### ISSUE THREAD

**Issue by julian-ladjani** [Liinkiing/discord-blindtest-bot#2]
When do a !bt help
Bot not answer and some errors appears on console: 
```
(node:28188) UnhandledPromiseRejectionWarning:   TypeError: fields.flat is not a function

  - MessageEmbed.js:450 Function.normalizeFields
    [discord-blindtest-bot]/[discord.js]/src/structures/MessageEmbed.js:450:8

  - MessageEmbed.js:266 MessageEmbed.addFields
    [discord-blindtest-bot]/[discord.js]/src/structures/MessageEmbed.js:266:42

  - MessageEmbed.js:257 MessageEmbed.addField
    [discord-blindtest-bot]/[discord.js]/src/structures/MessageEmbed.js:257:17

  - embeds.ts:45 Object.exports.INFO_EMBED
    /discord-bot/discord-blindtest-bot/src/utils/embeds.ts:45:6

  - blindtest.ts:33 BlindtestCommand.value
    /discord-bot/discord-blindtest-bot/src/commands/blindtest.ts:33:12

  - blindtest.ts:132 BlindtestCommand.value
    /discord-bot/discord-blindtest-bot/src/commands/blindtest.ts:132:24

  - blindtest.ts:67 BlindtestCommand.<anonymous>
    /discord-bot/discord-blindtest-bot/src/commands/blindtest.ts:67:14

  - blindtest.ts:57 step
    /discord-bot/discord-blindtest-bot/src/commands/blindtest.ts:57:23

  - blindtest.ts:38 Object.next
    /discord-bot/discord-blindtest-bot/src/commands/blindtest.ts:38:53

  - blindtest.ts:32
    /discord-bot/discord-blindtest-bot/src/commands/blindtest.ts:32:71


(node:28188) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 10)
```

**Comment by Liinkiing** [comment]
Hi!

Thanks you for using it, let me tell you first, I'm not going to lie to you, I know it's missing some documentation or something to help setup the bot, because it was made first for my personal usage!

But I'll be glad to help you set up the bot, not in the Discord / Airtable setup (you have to know how to setup those things and have a proper Airtable API Key / Discord BOT Token), but I've added two more env key in the .env file that maybe were missing in your setup.

In total, you must provide those parameters to the bot to properly run :

```ini
BOT_TOKEN=iamsosecretthatnsawontcatchme # your discord bot token
AIRTABLE_API_KEY=yourapikey # your airtable api key
AIRTABLE_BASE_ID=yourbaseid # your airtable base id, see more in the airtable docs
AIRTABLE_SONGS_TABLE_NAME=yourairtabletablenameforsongs # the name of the table (spreadsheet) in airtable used to store the songs.
AIRTABLE_LEADERBOARD_TABLE_NAME=yourairtabletablenameforleaderboard # the name of the table (spreadsheet) in airtable used to store the leaderboard.
```

About the songs table and the leaderboard table, you can see what are the columns needed in the `SongMapper` class (which translate a record from airtable to a `Song` entity for the bot) and same for the `EntryMapper`.

![image](https://user-images.githubusercontent.com/17753997/98997199-6bada000-2534-11eb-98b4-bad630fd6354.png)
here is an exemple of a correctly setup table. The code from the repo does not help you for getting those values, for this some work on your side is needed, I will not provide how I did it (but basically it consists of some farming of the Spotify API :trollface: ) 

There is at least a scripts (in the `scripts` folder) which is a very small node script to batch insert some data to airtable (again, you have to provide your own data source, because it is not relevant here in the repo)

I hope these guidelines will help you, I'll pin the issue maybe for someone else if they have trouble to properly setup the bot.

**Comment by julian-ladjani** [comment]
Hi, after configuration, add track... this problem is not resolved, can you contact me on discord julianjl#3664 when you have time ?

**Comment by Liinkiing** [comment]
> Hi, after configuration, add track... this problem is not resolved, can you contact me on discord julianjl#3664 when you have time ?

Sure 👍

**Comment by Liinkiing** [comment]
It was the Node version that was causing the issue. `.flat` method is available starting from Node v11
